.github/workflows/fcos-podman-next-build.yml

```diff
@@ -32,6 +33,10 @@ jobs:
       run: |
         pip3 install git+https://github.com/packit/wait-for-copr.git@main
 
+    - name: Set up cosa
```

Comment:
We shouldn't need cosa for this.

So here's how I think this would go roughly:
- Based on some trigger (e.g. polling new RPMs out of COPR, or there's a new quay.io/fedora/fedora-coreos:stable image published), start this job
- Do a `podman build` of an image that does `FROM: quay.io/fedora/fedora-coreos:stable`, and e.g. removes moby, adds the podman RPMs, adds subscription manager, etc...
- Run `osbuild` to convert the container image into disk images using the same manifests we use (these are currently hosted e.g. [here](https://github.com/coreos/coreos-assembler/blob/main/src/coreos.osbuild.x86_64.mpp.yaml) and [here](https://github.com/coreos/coreos-assembler/blob/main/src/coreos.osbuild.aarch64.mpp.yaml), but I think the goal is to move them out). This requires running on a system with full root privileges. For aarch64, [GHA was planning to add runners](https://github.blog/changelog/2023-10-30-accelerate-your-ci-cd-with-arm-based-hosted-runners-in-github-actions/) but it's not clear if that has happened yet. Alternatively, you can have a self-hosted runner on e.g. AWS. Nested virt shouldn't be required AIUI since osbuild uses loopbacks instead.
- Bundle up the disk images into container images and push to the registry

For the triggering, probably the simplest is polling and comparing the latest versions of the RPMs and base FCOS image against the latest pushed podman machine images (e.g. you can store the versioning in labels).

Reply:
/cc @dustymabe @baude

Reply:
so basically step 3 turns into "run this script".

Reply:
> We shouldn't need cosa for this.

Ack
 
> * Based on some trigger (e.g. polling new RPMs out of COPR, or there's a new quay.io/fedora/fedora-coreos:stable image published), start this job
> * Do a `podman build` of an image that does `FROM: quay.io/fedora/fedora-coreos:stable`, and e.g. removes moby, adds the podman RPMs, adds subscription manager, etc...

These two steps are pretty much happening already using https://github.com/containers/podman/blob/main/contrib/podman-next/fcos-podmanimage/Containerfile . This was taken from https://github.com/coreos/layering-examples/tree/main/podman-next .

subscription-manager addition isn't being done yet, I'll defer to @baude on that.

The container images built are pushed to https://quay.io/repository/podman/fcos?tab=tags . 


> * Run `osbuild` to convert the container image into disk images using the same manifests we use (these are currently hosted e.g. [here](https://github.com/coreos/coreos-assembler/blob/main/src/coreos.osbuild.x86_64.mpp.yaml) and [here](https://github.com/coreos/coreos-assembler/blob/main/src/coreos.osbuild.aarch64.mpp.yaml), but I think the goal is to move them out). This requires running on a system with full root privileges. For aarch64, [GHA was planning to add runners](https://github.blog/changelog/2023-10-30-accelerate-your-ci-cd-with-arm-based-hosted-runners-in-github-actions/) but it's not clear if that has happened yet. Alternatively, you can have a self-hosted runner on e.g. AWS. Nested virt shouldn't be required AIUI since osbuild uses loopbacks instead.


> * Bundle up the disk images into container images and push to the registry

Ack, thanks. I'll look into these two.

 
> For the triggering, probably the simplest is polling and comparing the latest versions of the RPMs and base FCOS image against the latest pushed podman machine images (e.g. you can store the versioning in labels).

Ack. The trigger for the existing action right now is a commit to the main branch of podman followed by a successful rpm build at the podman-next COPR. This is done using https://github.com/packit/wait-for-copr.git . So far, we haven't been checking for image updates at quay.io/fedora/fedora-coreos:stable though IIUC that's part of upcoming work.